### DISCUSSION TITLE
How to use JWK key for openid-client request?

### DISCUSSION THREAD

**Discussion by ush189** [panva/jose#98] [Q&A]
I want to get an access token with the [openid-client](https://www.npmjs.com/package/openid-client) module. It expects my private key to be formatted as a JWK key. But when I create one with this library they seem to be incompatible: 

```javascript
import { Issuer } from 'openid-client';
import { JWK } from 'jose';

public getToken(privateKey) {
    const key = JWK.asKey(privateKey);

    Issuer.discover(
        'https://keycloak.domain/auth/realms/realm-name'
    ).then(
        (issuer) => {
            const client = new issuer.Client(
                {
                    client_id: '<client-id>',
                    token_endpoint_auth_method: 'private_key_jwt',
                },
                {
                    keys: [key], // <-- types incompatible
                }
            );
             client
                .grant({ grant_type: 'client_credentials' })
                .then(token => {
                    console.log('token', token);
                }
            );
        }
    );
)
```

Am I doing it wrong?

**Comment by panva** [comment]
> Am I doing it wrong?

Yes.

You're supposed to pass a JWK formatted key, not a jose.JWK.Key representation of a key. Specifically so that the libraries aren't interlinked on the public API surface.

Use https://github.com/panva/jose/blob/v2.0.2/docs/README.md#keytojwkprivate to get a plain javascript JWK-formatted object.
